File: YOUR_ERP_CORE/frontend/pages/crm_customers.py
from frontend.pages.layout import base_layout


def crm_customers_page():
    content = """
<div class="page-header">
    <div style="display:flex;align-items:flex-start;justify-content:space-between;flex-wrap:wrap;gap:1rem;">
        <div>
            <h1>Clientes</h1>
            <p>Gestiona la cartera de clientes de tu empresa</p>
        </div>
        <div style="display:flex;gap:0.75rem;align-items:center;flex-wrap:wrap;">
            <a href="/app/crm" class="btn btn-ghost">&#8592; Pipeline</a>
            <button class="btn btn-primary" onclick="openCreateModal()">&#43; Nuevo Cliente</button>
        </div>
    </div>
</div>

<!-- Stats row -->
<div class="cards-row" id="cust-stats-row">
    <div class="stat-card">
        <div class="label">Total clientes</div>
        <div class="value" id="cust-stat-total">—</div>
        <div class="sub">registrados</div>
    </div>
    <div class="stat-card">
        <div class="label">Con oportunidades</div>
        <div class="value" id="cust-stat-with">—</div>
        <div class="sub">tienen al menos 1 lead</div>
    </div>
    <div class="stat-card">
        <div class="label">Sin oportunidades</div>
        <div class="value" id="cust-stat-without">—</div>
        <div class="sub">sin leads asignados</div>
    </div>
</div>

<!-- Search + Table -->
<div class="card" style="padding:0;overflow:hidden;">
    <div style="padding:1rem 1.25rem;border-bottom:1px solid #334155;display:flex;gap:0.75rem;align-items:center;">
        <input type="text" id="cust-search"
               placeholder="Buscar por nombre o RUT&hellip;"
               oninput="filterCustomers(this.value)"
               style="flex:1;background:#0f172a;border:1px solid #334155;border-radius:8px;padding:0.5rem 0.85rem;color:#e2e8f0;font-size:0.85rem;">
        <span id="cust-count-label" class="text-muted text-sm"></span>
    </div>
    <div class="table-wrap">
        <table class="data-table">
            <thead>
                <tr>
                    <th>Nombre</th>
                    <th>RUT / Tax&nbsp;ID</th>
                    <th>Direcci&oacute;n</th>
                    <th style="text-align:center;">Mandantes</th>
                    <th style="text-align:center;">Oportunidades</th>
                    <th style="text-align:right;">Acciones</th>
                </tr>
            </thead>
            <tbody id="cust-tbody">
                <tr><td colspan="6" style="text-align:center;padding:2.5rem;color:#64748b;">Cargando&hellip;</td></tr>
            </tbody>
        </table>
    </div>
</div>


<!-- ═══ MODAL: CREATE / EDIT CUSTOMER ═══════════════════════════ -->
<div id="cust-modal" class="modal-backdrop" style="display:none;" onclick="closeCustModalBackdrop(event)">
    <div class="modal-box modal-sm" onclick="event.stopPropagation()">
        <div class="modal-header">
            <h3 id="cust-modal-title">Nuevo Cliente</h3>
            <button class="modal-close" onclick="closeCustModal()">&#10005;</button>
        </div>
        <div class="modal-body">
            <input type="hidden" id="cust-id">
            <div class="form-row">
                <div class="form-group" style="flex:2;">
                    <label>Nombre *</label>
                    <input type="text" id="cust-name" placeholder="TechCorp Ltda" autocomplete="off">
                </div>
                <div class="form-group" style="flex:1;">
                    <label>RUT / Tax&nbsp;ID</label>
                    <input type="text" id="cust-taxid" placeholder="76.123.456-7">
                </div>
            </div>
            <div class="form-row">
                <div class="form-group" style="flex:2;">
                    <label>Direcci&oacute;n</label>
                    <input type="text" id="cust-address" placeholder="Av. Providencia 1234">
                </div>
                <div class="form-group" style="flex:1;">
                    <label>Ciudad</label>
                    <input type="text" id="cust-city" placeholder="Santiago">
                </div>
            </div>
            <div class="form-group">
                <label>Documentos especificos de acreditacion</label>
                <select id="cust-acc-template-selector" multiple size="6"></select>
                <span class="text-muted text-sm">Se suman a los documentos estandar para este cliente y luego aplican automaticamente a los trabajadores asignados.</span>
            </div>
        </div>
        <div class="modal-footer">
            <button class="btn btn-danger" id="cust-delete-btn"
                    onclick="deleteCustomer()" style="display:none;margin-right:auto;">
                &#128465; Eliminar
            </button>
            <button class="btn btn-ghost" onclick="closeCustModal()">Cancelar</button>
            <button class="btn btn-primary" onclick="saveCust()">Guardar</button>
        </div>
    </div>
</div>

<!-- ═══ MODAL: ADD MANDANTE ════════════════════════════════════ -->
<div id="mandante-modal" class="modal-backdrop" style="display:none;" onclick="closeMandanteModalBackdrop(event)">
    <div class="modal-box modal-sm" onclick="event.stopPropagation()">
        <div class="modal-header">
            <h3>Nuevo Mandante</h3>
            <button class="modal-close" onclick="closeMandanteModal()">&#10005;</button>
        </div>
        <div class="modal-body">
            <input type="hidden" id="mand-customer-id">
            <div class="form-group">
                <label>Nombre Completo *</label>
                <input type="text" id="mand-name" placeholder="Ej: Juan Pérez" autocomplete="off">
            </div>
            <div class="form-group">
                <label>Cargo</label>
                <input type="text" id="mand-position" placeholder="Ej: Gerente de Finanzas">
            </div>
            <div class="form-row">
                <div class="form-group" style="flex:1;">
                    <label>Email</label>
                    <input type="email" id="mand-email" placeholder="correo@empresa.cl">
                </div>
                <div class="form-group" style="flex:1;">
                    <label>Teléfono</label>
                    <input type="text" id="mand-phone" placeholder="+56 9 1234 5678">
                </div>
            </div>
        </div>
        <div class="modal-footer">
            <button class="btn btn-ghost" onclick="closeMandanteModal()">Cancelar</button>
            <button class="btn btn-primary" onclick="saveMandante()">Guardar Contacto</button>
        </div>
    </div>
</div>
"""
    return base_layout(
        title="Clientes — CRM",
        page_id="crm-customers",
        content=content,
        scripts=["crm_customers.js"]
    )

File: YOUR_ERP_CORE/frontend/static/js/crm_customers.js
let CUSTS = {
    all: [],
    filtered: [],
};
let ACCREDITATION_TEMPLATES = [];

document.addEventListener('DOMContentLoaded', async () => {
    if (!API.getToken()) { window.location.href = '/app/login'; return; }
    highlightNav('/app/crm');
    await Promise.all([loadCustomers(), loadAccreditationTemplates()]);
});

async function loadAccreditationTemplates() {
    const res = await API.get('/hr/accreditation/templates');
    ACCREDITATION_TEMPLATES = res?.data?.results || [];
    fillCustomerTemplateSelector([]);
}

async function loadCustomers() {
    const res = await API.get('/crm/customers');
    if (!res || !res.success) {
        document.getElementById('cust-tbody').innerHTML =
            '<tr><td colspan="6" style="text-align:center;padding:2rem;color:#ef4444;">Error cargando clientes</td></tr>';
        return;
    }
    CUSTS.all = res.data.results || [];
    CUSTS.filtered = [...CUSTS.all];
    updateStats();
    renderTable(CUSTS.filtered);
}

function updateStats() {
    const total = CUSTS.all.length;
    const withL = CUSTS.all.filter((c) => (c.lead_count || 0) > 0).length;
    setText('cust-stat-total', total);
    setText('cust-stat-with', withL);
    setText('cust-stat-without', total - withL);
}

function filterCustomers(q) {
    const s = (q || '').toLowerCase();
    CUSTS.filtered = s
        ? CUSTS.all.filter((c) =>
            String(c.name || '').toLowerCase().includes(s) ||
            String(c.tax_id || '').toLowerCase().includes(s))
        : [...CUSTS.all];
    renderTable(CUSTS.filtered);
}

function renderTable(list) {
    const tbody = document.getElementById('cust-tbody');
    const label = document.getElementById('cust-count-label');
    if (label) label.textContent = `${list.length} de ${CUSTS.all.length}`;

    if (!list.length) {
        tbody.innerHTML = '<tr><td colspan="6" style="text-align:center;padding:2.5rem;color:#64748b;">Sin resultados</td></tr>';
        return;
    }
    tbody.innerHTML = list.map((c) => `
        <tr>
            <td><strong style="color:#f1f5f9;">${escHtml(c.name)}</strong></td>
            <td class="text-muted">${escHtml(c.tax_id || '-')}</td>
            <td class="text-muted" style="max-width:180px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;">${escHtml(c.address || '-')}</td>
            <td style="text-align:center;">
                <span class="badge ${c.mandante_count > 0 ? 'badge-info' : ''}" style="color:${c.mandante_count > 0 ? '#38bdf8' : '#64748b'};background:transparent;border:1px solid ${c.mandante_count > 0 ? '#38bdf8' : '#334155'};">
                    ${c.mandante_count || 0}
                </span>
            </td>
            <td style="text-align:center;">
                ${(c.lead_count || 0) > 0
                    ? `<span class="badge badge-info" style="cursor:pointer;" onclick="goToPipeline(${c.id})" title="Ver en pipeline">${c.lead_count}</span>`
                    : '<span class="text-muted text-sm">0</span>'}
            </td>
            <td style="text-align:right;white-space:nowrap;">
                <button class="btn btn-ghost btn-sm" onclick="openMandanteModal(${c.id})" style="margin-right:0.25rem;color:#38bdf8;" title="Agregar contacto">+ Mandante</button>
                <button class="btn btn-ghost btn-sm" onclick="window.location.href='/app/crm/customers/${c.id}'" style="margin-right:0.25rem">Detalle</button>
                <button class="btn btn-ghost btn-sm" onclick="openEditModal(${c.id})">Editar</button>
            </td>
        </tr>`).join('');
}

function goToPipeline() {
    window.location.href = '/app/crm';
}

function _fillCustModal(c) {
    document.getElementById('cust-id').value = c.id || '';
    document.getElementById('cust-name').value = c.name || '';
    document.getElementById('cust-taxid').value = c.tax_id || '';
    document.getElementById('cust-address').value = c.address || '';
    document.getElementById('cust-city').value = c.city || '';
}

function fillCustomerTemplateSelector(selectedCodes = []) {
    const select = document.getElementById('cust-acc-template-selector');
    if (!select) return;
    const selected = new Set((selectedCodes || []).map(item => String(item)));
    select.innerHTML = ACCREDITATION_TEMPLATES.map((item) => `
        <option value="${escHtml(item.code)}" ${selected.has(String(item.code)) ? 'selected' : ''}>
            ${escHtml(item.name)}
        </option>
    `).join('');
}

function getSelectedTemplateCodes() {
    const select = document.getElementById('cust-acc-template-selector');
    if (!select) return [];
    return Array.from(select.selectedOptions).map(option => option.value).filter(Boolean);
}

function openCreateModal() {
    document.getElementById('cust-modal-title').textContent = 'Nuevo cliente';
    _fillCustModal({});
    fillCustomerTemplateSelector([]);
    document.getElementById('cust-delete-btn').style.display = 'none';
    document.getElementById('cust-modal').style.display = 'flex';
    setTimeout(() => document.getElementById('cust-name').focus(), 80);
}

async function openEditModal(id) {
    const c = CUSTS.all.find((item) => item.id === id);
    if (!c) return;
    document.getElementById('cust-modal-title').textContent = 'Editar cliente';
    _fillCustModal(c);
    document.getElementById('cust-delete-btn').style.display = '';
    document.getElementById('cust-modal').style.display = 'flex';
    const res = await API.get(`/hr/accreditation/customers/${id}/requirements`);
    fillCustomerTemplateSelector(res?.data?.selected_codes || []);
    setTimeout(() => document.getElementById('cust-name').focus(), 80);
}

function closeCustModal() {
    document.getElementById('cust-modal').style.display = 'none';
}

function closeCustModalBackdrop(event) {
    if (event.target === document.getElementById('cust-modal')) closeCustModal();
}

async function saveCust() {
    const id = document.getElementById('cust-id').value;
    const name = document.getElementById('cust-name').value.trim();
    if (!name) { showToast('El nombre es obligatorio', 'error'); return; }

    const payload = {
        name,
        tax_id: document.getElementById('cust-taxid').value.trim(),
        address: document.getElementById('cust-address').value.trim(),
        city: document.getElementById('cust-city').value.trim(),
    };

    const res = id
        ? await API.put(`/crm/customers/${id}`, payload)
        : await API.post('/crm/customers', payload);

    if (res && res.success) {
        const customerId = res.data?.id || id;
        await API.put(`/hr/accreditation/customers/${customerId}/requirements`, {
            template_codes: getSelectedTemplateCodes(),
        });
        showToast(id ? 'Cliente actualizado' : 'Cliente creado', 'success');
        closeCustModal();
        await loadCustomers();
    } else {
        showToast((res?.errors || ['Error']).join(', '), 'error');
    }
}

async function deleteCustomer() {
    const id = document.getElementById('cust-id').value;
    if (!id) return;
    const c = CUSTS.all.find((item) => item.id === Number(id));
    if (!confirm(`Eliminar cliente "${c?.name || ''}"?\nSus oportunidades quedaran sin cliente asignado.`)) return;

    const res = await API.del(`/crm/customers/${id}`);
    if (res && res.success) {
        showToast('Cliente eliminado', 'success');
        closeCustModal();
        await loadCustomers();
    } else {
        showToast('Error al eliminar', 'error');
    }
}

function openMandanteModal(customerId) {
    document.getElementById('mand-customer-id').value = customerId;
    document.getElementById('mand-name').value = '';
    document.getElementById('mand-position').value = '';
    document.getElementById('mand-email').value = '';
    document.getElementById('mand-phone').value = '';
    document.getElementById('mandante-modal').style.display = 'flex';
    setTimeout(() => document.getElementById('mand-name').focus(), 80);
}

function closeMandanteModal() {
    document.getElementById('mandante-modal').style.display = 'none';
}

function closeMandanteModalBackdrop(event) {
    if (event.target === document.getElementById('mandante-modal')) {
        closeMandanteModal();
    }
}

async function saveMandante() {
    const customerId = document.getElementById('mand-customer-id').value;
    const name = document.getElementById('mand-name').value.trim();
    if (!name) { showToast('El nombre es obligatorio', 'error'); return; }

    const payload = {
        name,
        position: document.getElementById('mand-position').value.trim(),
        email: document.getElementById('mand-email').value.trim(),
        phone: document.getElementById('mand-phone').value.trim(),
    };

    const btn = document.querySelector('#mandante-modal .btn-primary');
    if (btn) btn.disabled = true;

    const res = await API.post(`/crm/customers/${customerId}/mandantes`, payload);

    if (btn) btn.disabled = false;

    if (res && res.success) {
        showToast('Contacto B2B creado exitosamente', 'success');
        closeMandanteModal();
        await loadCustomers();
    } else {
        showToast((res?.errors || ['Error desconocido']).join(', '), 'error');
    }
}

document.addEventListener('keydown', (e) => {
    if (e.key === 'Escape') closeCustModal();
});

function highlightNav(path) {
    document.querySelectorAll('.sidebar nav a').forEach((a) =>
        a.classList.toggle('active', a.getAttribute('href') === path));
}

function setText(id, val) {
    const el = document.getElementById(id);
    if (el) el.textContent = val;
}

function escHtml(str) {
    return String(str || '')
        .replace(/&/g, '&amp;')
        .replace(/</g, '&lt;')
        .replace(/>/g, '&gt;')
        .replace(/"/g, '&quot;');
}
